### PR TITLE
fix: check git source when repo type is `git`

### PR DESCRIPTION
**Pull request by vLia**
## Pull request description 

During investigation on https://github.com/kubeshop/testkube/issues/3198, I saw that we do not have the option to specify the git type (git-file or git-dir) when we are using test sources via the UI.

This PR addresses that by adding a function to decide on the go what the type of the input is. Unfortunately it is not possible to get the type of a path without cloning the repo, so this means the repo will be cloned twice on execution: once in the api-server to check the type on the go, and once in the init-container. This has the upside that in case the type changes between the creation of the test and the execution, Testkube will be aware of it.

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [x] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [x] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: pkg/scheduler/test_scheduler.go [hunk 1]
```diff
@@ -14,12 +14,15 @@ import (
...
 	"github.com/kubeshop/testkube/pkg/executor/client"
+	"github.com/kubeshop/testkube/pkg/executor/content"
 	testsmapper "github.com/kubeshop/testkube/pkg/mapper/tests"
 	"github.com/kubeshop/testkube/pkg/workerpool"
 )
 
 const (
 	containerType = "container"
+	// tempDir is the temporary directory used to evaluate a type of a 'git' source
+	tempDir = "/tmp"
```
Comment (vsukhin) [inline pkg/scheduler/test_scheduler.go:25]: you can use it like I did for git check

			dir, err := os.MkdirTemp("", "checkout")
			if err != nil {
				return s.Error(c, http.StatusBadGateway, err)
			}
			defer os.RemoveAll(dir) // clean up

File: pkg/scheduler/test_scheduler.go [hunk 2]
```diff
@@ -247,6 +250,19 @@ func (s *Scheduler) getExecuteOptions(namespace, id string, request testkube.Exe
...
 
 		testCR.Spec = mergeContents(testCR.Spec, testSourceCR.Spec)
+
+		if testSourceCR.Spec.Type_ == "" && testSourceCR.Spec.Repository.Type_ == "git" {
+			fetcher := content.NewFetcher(tempDir)
+			content := testsmapper.MapTestContentFromSpec(testCR.Spec.Content)
+			t, err := fetcher.CalculateGitContentType(*content.Repository)
+			if err == nil {
+				testCR.Spec.Content.Type_ = t
+				testCR.Spec.Content.Repository.Type_ = t
```
Comment (vsukhin) [inline pkg/scheduler/test_scheduler.go:260]: guess, we use `git` for this now

File: pkg/scheduler/test_scheduler.go (hunk 1 above)
Comment (vsukhin) [reply pkg/scheduler/test_scheduler.go:25]: Or os.GetTemp()

File: pkg/executor/content/fetcher.go [hunk 3]
```diff
@@ -187,3 +189,37 @@ func (f Fetcher) configureUseOfCertificate() error {
...
 	return nil
 }
+
+// CalculateGitContentType returns the type of the git test source
+func (f Fetcher) CalculateGitContentType(repo testkube.Repository) (string, error) {
+	if repo.Uri == "" || repo.Path == "" {
+		return "", errors.New("repository uri and path should be populated")
+	}
+
+	dir, err := os.MkdirTemp("/tmp", "temp-git-files")
```
Comment (vsukhin) [inline pkg/executor/content/fetcher.go:199]: os.GetTemp()

**Review by vsukhin — APPROVED** [review]
some miinor comments

File: pkg/scheduler/test_scheduler.go (hunk 2 above)
Comment (vLia) [reply pkg/scheduler/test_scheduler.go]: removed this line, tested with git-file and git-dir, still works

File: pkg/executor/content/fetcher.go (hunk 3 above)
Comment (vLia) [reply pkg/executor/content/fetcher.go]: updated to empty string and let os.MkdirTemp(..) call this function

**Pull request merged**